### ISSUE TITLE
Failed building libstm32cubewb.a and libsubghz.a

### ISSUE THREAD

**Issue by ImprovingRigmarole** [flipperdevices/flipperzero-firmware#1483]
### Describe the bug.

Follow up of #1481 on Windows 10.

I started running `fbt.cmd` after checking out from `release` branch (currently [3e5d499](https://github.com/flipperdevices/flipperzero-firmware/commit/3e5d499b443f86f9313aaaa75b9925e59d4a137a)) and `fbt.cmd` didn't install anything.

After switching to `dev` (c40e8811d68e9f4b8f603ae5d5826b814521014d), `fbt.cmd` successully installed its toolchain,
then succesfully compiling a bunch of .c files, and when it arrives at calling `arm-none-eabi-ar` to link .a files there seems to be some argument badly formatted and gives error, then `ar`'s help for each .a it didn't take.

### Reproduction

- clone repo 
- checkout c40e8811d68e9f4b8f603ae5d5826b814521014d  (current `dev`)
- run `fbt.cmd`

### Logs

```Text
//////// first run of fbt.cmd with build folder empty
//////// everything above this is only successful CC compiling
        CC      lib\FreeRTOS-Kernel\stream_buffer.c
        CC      lib\FreeRTOS-Kernel\tasks.c
        CC      lib\FreeRTOS-Kernel\timers.c
        CC      lib\FreeRTOS-Kernel\portable\GCC\ARM_CM4F\port.c
        CC      lib\microtar\src\microtar-stdio.c
        AR      build\f7-firmware-D\STM32CubeWB\libstm32cubewb.a
        CC      lib\microtar\src\microtar.c
        CC      lib\toolbox\stream\buffered_file_stream.c
        CC      lib\toolbox\stream\file_stream.c
        CC      lib\toolbox\stream\stream.c
        CC      lib\toolbox\stream\stream_cache.c
arm-none-eabi-ar: invalid option -- @
Usage: arm-no   CC      lib\toolbox\stream\string_stream.c
ne-eabi-ar      AR      build\f7-firmware-D\microtar\libmicrotar.a
 [em    AR      build\f7-firmware-D\freertos\libfreertos.a
ulation options] [-]{dmpqrstx}[abcDfilMNoOPsSTuvV] [--plugin <nam   CC      lib\toolbox\tar\tar_archive.c
e       CC      lib\toolbox\args.c
>] [member-name] [count] ar     CC      lib\toolbox\crc32_calc.c
chive-file file..       CC      lib\toolbox\dir_walk.c
.
        CC      lib\toolbox\hex.c
        CC      lib\toolbox\hmac_sha256.c
arm-no     CC      lib\toolbox\manchester_decoder.c
ne-ea   CC      lib\toolbox\manchester_encoder.c
bi-ar -M [<mri-script]
 commands:
  d            - delete file(s) from the archive
        CC      lib\toolbox\md5.c
        CC      lib\toolbox\path.c
m[      CC      lib\toolbox\random_name.c
        CC      lib\toolbox\saved_struct.c
ab      CC      lib\toolbox\sha256.c
]           CC      lib\toolbox\value_index.c
-       VERSION build\f7-firmware-D\toolbox\version.inc.h
move file(s) in the archive
  p     CC      lib\ST25RFAL002\source\st25r3916\rfal_rfst25r3916.c
           - print file(s) fo   CC      lib\ST25RFAL002\source\st25r3916\st25r3916.c
un      CC      lib\ST25RFAL002\source\st25r3916\st25r3916_aat.c
d in    CC      lib\ST25RFAL002\source\st25r3916\st25r3916_com.c
the  CC      lib\ST25RFAL002\source\st25r3916\st25r3916_irq.c
 ar     CC      lib\ST25RFAL002\source\st25r3916\st25r3916_led.c
chive
  q[f]         - quick append file(s) to the archive
  r[ab][f][u]  - replace existing or insert new file(s) into the archive
  s            - act as ranlib
  t[O][v]      - display contents of the archive
  x[o]         - extract file(s) from the archive
 comma  CC      lib\ST25RFAL002\source\custom_analog_config.c
        CC      lib\ST25RFAL002\source\rfal_analogConfig.c
n       CC      lib\ST25RFAL002\source\rfal_crc.c
d       CC      lib\ST25RFAL002\source\rfal_dpo.c
 sp     CC      lib\ST25RFAL002\source\rfal_iso15693_2.c
ecific modifiers:
  [a]          - put file(s) a  CC      lib\ST25RFAL002\source\rfal_isoDep.c
fter    CC      lib\ST25RFAL002\source\rfal_nfc.c
[me     CC      lib\ST25RFAL002\source\rfal_nfcDep.c
mber-   CC      lib\ST25RFAL002\source\rfal_nfca.c
na      CC      lib\ST25RFAL002\source\rfal_nfcb.c
m       CC      lib\ST25RFAL002\source\rfal_nfcf.c
e]
 Version information updated
 [b]          - put file(s) before [member-name] (same as [i])
        CC      lib\toolbox\version.c
  [D]          - use zero for timestamps and uids/gids
  [U]          - use actual timestamps and uids/gids (default)
        AR      build\f7-firmware-D\toolbox\libtoolbox.a
  [N]          - use instance [count] of name
  [f]           CC      lib\ST25RFAL002\source\rfal_nfcv.c
     -  CC      lib\ST25RFAL002\source\rfal_picopass.c
truncate        CC      lib\ST25RFAL002\source\rfal_st25tb.c
 ins    CC      lib\ST25RFAL002\source\rfal_st25xv.c
erted   CC      lib\ST25RFAL002\source\rfal_t1t.c
 fi     CC      lib\ST25RFAL002\source\rfal_t2t.c
l       CC      lib\ST25RFAL002\source\rfal_t4t.c
e names
  [P]          - use full path names when matching
  [o]          - preserve original dates
  [O]          - displ  CC      lib\ST25RFAL002\platform.c
ay offsets of files in the archive
        CC      lib\ST25RFAL002\timer.c
[u      CC      lib\libusb_stm32\src\usbd_core.c
]       CC      lib\libusb_stm32\src\usbd_stm32wb55_devfs.c
        CC      lib\drivers\bq25896.c
 - onl  CC      lib\drivers\bq27220.c
y       CC      lib\drivers\cc1101.c
 repla  CC      lib\drivers\lp5562.c
        AR      build\f7-firmware-D\ST25RFAL002\libst25rfal002.a
ce      CC      lib\fatfs\option\unicode.c
files that are newer than current archive con   CC      lib\fatfs\diskio.c
t       AR      build\f7-firmware-D\libusb_stm32\libusb_stm32.a
e       CC      lib\fatfs\ff.c
nts
        CC      lib\fatfs\ff_gen_drv.c
 generic modifiers:
        CC      lib\flipper_format\flipper_format.c
        CC      lib\flipper_format\flipper_format_stream.c
[c]             CC      lib\infrared\encoder_decoder\common\infrared_common_decoder.c
        CC      lib\infrared\encoder_decoder\common\infrared_common_encoder.c
- do not warn if the library had to be created
        CC      lib\infrared\encoder_decoder\common\infrared_common_protocol_defs.c
 [s]    AR      build\f7-firmware-D\drivers\libhwdrivers.a
       - create an archive index (cf. ranlib)
  [l <text> ]  - specify the dependencies of this library
  [S]          - do not build a symbol table
  [T]          - make a thin archive
  [v]          - be verbose
  [V]          - display the version    AR      build\f7-firmware-D\flipper_format\libflipperformat.a
number
  @<file>      - read options from <file>
  --target=BFDNAME - specify the target object format as BFDNAME
  --output=DIRNAME - specify the output directory for extraction operations
  --record-libdeps=<text> - specify the dependencies of this library
 optional:
  --plugin <p> - load the specified plugin
 emulation options:
  No emulation specific options
arm-none-eabi-ar: supported targets: elf32-littlearm elf32-littlearm-fdpic elf32-bigarm elf32-bigarm-fdpic elf32-little elf32-big srec symbolsrec verilog tekhex binary ihex plugin
        CC      lib\infrared\encoder_decoder\nec\infrared_decoder_nec.c
scons: *** [build\f7-firmware-D\STM32CubeWB\libstm32cubewb.a] Error 1
********** ERRORS **********
Failed building build\f7-firmware-D\STM32CubeWB\libstm32cubewb.a: Error 1

//////// re running fbt.cmd after this
        AR      build\f7-firmware-D\STM32CubeWB\libstm32cubewb.a
arm-none-eabi-ar: invalid option -- @
Usage: arm-none-eabi-ar [emulation options] [-]{dmpqrstx}[abcDfilMNoOPsSTuvV] [--plugin <name>] [member-name] [count] archive-file file...
       arm-none-eabi-ar -M [<mri-script]
 commands:
  d            - delete file(s) from the archive
  m[ab]        - move file(s) in the archive
  p            - print file(s) found in the archive
  q[f]         - quick append file(s) to the archive
  r[ab][f][u]  - replace existing or insert new file(s) into the archive
  s            - act as ranlib
  t[O][v]      - display contents of the archive
  x[o]         - extract file(s) from the archive
 command specific modifiers:
  [a]          - put file(s) after [member-name]
  [b]          - put file(s) before [member-name] (same as [i])
  [D]          - use zero for timestamps and uids/gids
  [U]          - use actual timestamps and uids/gids (default)
  [N]          - use instance [count] of name
  [f]          - truncate inserted file names
  [P]          - use full path names when matching
  [o]          - preserve original dates
  [O]          - display offsets of files in the archive
  [u]          - only replace files that are newer than current archive contents
 generic modifiers:
  [c]          - do not warn if the library had to be created
  [s]          - create an archive index (cf. ranlib)
  [l <text> ]  - specify the dependencies of this library
  [S]          - do not build a symbol table
  [T]          - make a thin archive
  [v]          - be verbose
  [V]          - display the version number
  @<file>      - read options from <file>
  --target=BFDNAME - specify the target object format as BFDNAME
  --output=DIRNAME - specify the output directory for extraction operations
  --record-libdeps=<text> - specify the dependencies of this library
 optional:
  --plugin <p> - load the specified plugin
 emulation options:
  No emulation specific options
arm-none-eabi-ar: supported targets: elf32-littlearm elf32-littlearm-fdpic elf32-bigarm elf32-bigarm-fdpic elf32-little elf32-big srec symbolsrec verilog tekhex binary ihex plugin
        AR      build\f7-firmware-D\fatfs\libfatfs.a
        CC      lib\infrared\encoder_decoder\nec\infrared_encoder_nec.c
        CC      lib\infrared\encoder_decoder\nec\infrared_nec_spec.c
        CC      lib\infrared\encoder_decoder\rc5\infrared_decoder_rc5.c
        CC      lib\infrared\encoder_decoder\rc5\infrared_encoder_rc5.c
        CC      lib\infrared\encoder_decoder\rc5\infrared_rc5_spec.c
scons: *** [build\f7-firmware-D\STM32CubeWB\libstm32cubewb.a] Error 1
        CC      lib\infrared\encoder_decoder\rc6\infrared_decoder_rc6.c
********** ERRORS **********
Failed building build\f7-firmware-D\STM32CubeWB\libstm32cubewb.a: Error 1
```

**Comment by ImprovingRigmarole** [comment]
ok I just checked what's happening with ProcMon, 
and it seems that some commands passed to `arm-none-eabi-ar` are correct (most) :
```
arm-none-eabi-ar  rc "build\f7-firmware-D\littlefs\liblittlefs.a" "build\f7-firmware-D\littlefs\littlefs\lfs.o" "build\f7-firmware-D\littlefs\littlefs\lfs_util.o"
```
but a few ones get some weird temp folder path :
```
arm-none-eabi-ar  @C:\WINDOWS\TEMP\tmpffimpor2.lnk
```
file which is created by python process launched with those args `python  lib\scons\scripts\scons.py -Q --warn=target-not-built`

and it's those which fail obviously...
now what could happen in the build scripts to get to this, I don't know it enough for now...

**Comment by ImprovingRigmarole** [comment]
ok problem fixed !!!

so after seeing the problem seemingly came from Scons, I went to check over Scons' issues, and found this :
https://github.com/SCons/scons/issues/2973#issue-285460151

so basically when scons need to pass a long list of `.o` files to archive in an `.a` file to `arm-none-eabi-ar`, it stores the long command in a file and uses the `@file` for `ar` to read it.

- Scons uses the python `tempfile` package for this, which uses the envvar `TMPDIR` to store those.
- In my case I didn't have any User envvar `TMPDIR` set, so it used my System `TMPDIR` envvar which was set to `C:\WINDOWS\TEMP`.

It worked well for the python part of `fbt` and `scons` to create the file there, but somehow `ar` couldn't access the file and as specified in `ar`'s doc :
> If file does not exist, or cannot be read, then the option will be treated literally, and not removed. 

so it just failed and wasn't able to create any `.a` files for this.
Setting the `TMPDIR` to my user temp folder solved the thing.

**TLDR** for anyone on Windows stumbling on the same problem : 
- if there isn't an User Environment Variable on your windows user account (which is very probable if you arrived here)
- set an User Environment Variable named `TMPDIR` to have the value `%USERPROFILE%\AppData\Local\Temp`
